From 2d1d8ae18caa2f3b54f9681bd61f1feb98e2882e Mon Sep 17 00:00:00 2001
From: Matheus Sampaio Queiroga <srherobrine20@gmail.com>
Date: Sat, 18 Jul 2026 22:03:40 +0000
Subject: [PATCH 119/160] net: airoha: do not treat GEM delineation loss as
 optical LOS

G_INT_STATUS bit 7 reports loss of GEM delineation. The vendor ISR
records the LCDG alarm but does not transition the activation state
machine. Optical LOS is reported separately by the xPON PHY and SFP
callbacks.

Calling ploam_notify_los() for this bit forces a valid O5 link into O6
and resets the MAC after TO2. Keep the interrupt diagnostic, but leave
optical state transitions to the actual link-down callback.

Signed-off-by: Matheus Sampaio Queiroga <srherobrine20@gmail.com>
---
 drivers/net/ethernet/airoha/airoha_xpon.c | 11 +++++------
 1 file changed, 5 insertions(+), 6 deletions(-)

diff --git a/drivers/net/ethernet/airoha/airoha_xpon.c b/drivers/net/ethernet/airoha/airoha_xpon.c
index d61ee4d6135e..85d548b421c9 100644
--- a/drivers/net/ethernet/airoha/airoha_xpon.c
+++ b/drivers/net/ethernet/airoha/airoha_xpon.c
@@ -428,7 +428,7 @@ static void airoha_xpon_phy_stop(struct device *dev, struct phy *phy,
 #define INT_RANGING_REQ_RECV	BIT(4)
 #define INT_SN_ONU_SEND_O4	BIT(5)
 #define INT_SN_REQ_CRS		BIT(6)
-#define INT_LOS_GEM_DEL		BIT(7)
+#define INT_LOSS_GEM_DEL	BIT(7)
 #define INT_AES_KEY_SWITCH_DONE	BIT(8)
 #define INT_TOD_UPDATE_DONE	BIT(9)
 #define INT_TOD_1PPS		BIT(10)
@@ -459,7 +459,7 @@ static void airoha_xpon_phy_stop(struct device *dev, struct phy *phy,
 #define GPON_INT_ACTIVATION_MASK	(INT_PLOAMD_RECV | INT_SN_REQ_RECV | \
 				 INT_SN_ONU_SEND_O3 | INT_RANGING_REQ_RECV | \
 				 INT_SN_ONU_SEND_O4 | INT_SN_REQ_CRS | \
-				 INT_LOS_GEM_DEL | INT_AES_KEY_SWITCH_DONE | \
+				 INT_LOSS_GEM_DEL | INT_AES_KEY_SWITCH_DONE | \
 				 INT_DYING_GASP | INT_CAL_GNT_SIZE_DONE)
 /*
  * The EN7523 vendor driver enables only the common receive/burst errors.
@@ -2159,10 +2159,9 @@ static void gpon_irq_work_fn(struct work_struct *work)
 			ploam_notify_dying_gasp(priv->ploam);
 		}
 
-		if (active & INT_LOS_GEM_DEL) {
-			dev_warn(priv->dev, "GPON LOS/GEM-delete interrupt\n");
-			ploam_notify_los(priv->ploam);
-		}
+		if (active & INT_LOSS_GEM_DEL)
+			dev_warn_ratelimited(priv->dev,
+					     "GPON loss of GEM delineation interrupt\n");
 
 		if (active & GPON_INT_ERROR_MASK)
 			gpon_dump_error_counters(priv, active &
-- 
2.55.0

